jax/experimental/jax2tf/tests/primitives_test.py

```diff
@@ -284,52 +282,51 @@ def test_select_and_gather_add(self, harness: primitive_harness.Harness):
         raise unittest.SkipTest("TODO: fix bug: nan vs not-nan")
-      if harness.params["dtype"] is np.float16 and jtu.device_under_test() == "tpu":
+      if dtype is np.float16 and jtu.device_under_test() == "tpu":
         raise unittest.SkipTest("TODO: fix bug: nans and infs")
 
-      # digamma is not defined at 0 and -1
-      def custom_assert(result_jax, result_tf):
-        # lax.digamma returns NaN and tf.math.digamma returns inf
-        special_cases = (arg == 0.) | (arg == -1.)
-        nr_special_cases = np.count_nonzero(special_cases)
-        self.assertAllClose(np.full((nr_special_cases,), dtype(np.nan)),
-                            result_jax[special_cases])
-        self.assertAllClose(np.full((nr_special_cases,), dtype(np.inf)),
-                            result_tf[special_cases])
-        # non-special cases are equal
-        self.assertAllClose(result_jax[~ special_cases],
-                            result_tf[~ special_cases])
+      # In the bfloat16 case, TF and lax both return NaN in undefined cases.
+      if not dtype is dtypes.bfloat16:
+        # digamma is not defined at 0 and -1
+        def custom_assert(result_jax, result_tf):
+          # lax.digamma returns NaN and tf.math.digamma returns inf
+          special_cases = (arg == 0.) | (arg == -1.)
+          nr_special_cases = np.count_nonzero(special_cases)
+          self.assertAllClose(np.full((nr_special_cases,), dtype(np.nan)),
+                              result_jax[special_cases])
+          self.assertAllClose(np.full((nr_special_cases,), dtype(np.inf)),
+                              result_tf[special_cases])
+          # non-special cases are equal
+          self.assertAllClose(result_jax[~ special_cases],
+                              result_tf[~ special_cases])
     if lax_name == "erf_inv":
-      # TODO(necula): fix bug with erf_inv/f16
-      if dtype is np.float16:
-        raise unittest.SkipTest("TODO: fix bug")
       # TODO(necula): fix erf_inv bug on TPU
       if jtu.device_under_test() == "tpu":
         raise unittest.SkipTest("erf_inv bug on TPU: nan vs non-nan")
-      # erf_inf is not defined for arg <= -1 or arg >= 1
-      def custom_assert(result_jax, result_tf):  # noqa: F811
-        # for arg < -1 or arg > 1
-        # lax.erf_inf returns NaN; tf.math.erf_inv return +/- inf
-        special_cases = (arg < -1.) | (arg > 1.)
-        nr_special_cases = np.count_nonzero(special_cases)
-        self.assertAllClose(np.full((nr_special_cases,), dtype(np.nan)),
-                            result_jax[special_cases])
-        signs = np.where(arg[special_cases] < 0., -1., 1.)
-        self.assertAllClose(np.full((nr_special_cases,), signs * dtype(np.inf)),
-                            result_tf[special_cases])
-        # non-special cases are equal
-        self.assertAllClose(result_jax[~ special_cases],
-                            result_tf[~ special_cases])
+      if not dtype in [np.float16, dtypes.bfloat16]:
```

Comment:
Please add comment as above that for float16 JAX and jax2tf return the same thing

Reply:
Done